tests/basic/sharedv4_svc_test.go

```diff
@@ -259,84 +266,303 @@ var _ = Describe("{Shared4SvcFailoverIO}", func() {
 			}
 		})
 
-		Step("for each context, restart volume driver on NFS server node and verify I/O", func() {
-			for _, ctx := range testSv4Contexts {
+		for _, ctx := range testSv4Contexts {
+			Step(fmt.Sprintf("induce multiple sharedv4 service failovers and verify I/O for app %s", ctx.App.Key), func() {
 				vols, err := Inst().S.GetVolumes(ctx)
 				Expect(err).NotTo(HaveOccurred())
 
 				apiVol, err := Inst().V.InspectVolume(vols[0].ID)
 				Expect(err).NotTo(HaveOccurred())
 
 				// verify failover and failback by repeating the steps below
-				for i := 0; i < 2; i++ {
+				for i := 0; i < 4; i++ {
 					var countersBefore, countersAfter map[string]appCounter
 					var attachedNodeBefore, attachedNodeAfter *node.Node
+					var failoverMethod string
 
-					// get the attached node before the failover
-					attachedNodeBefore, err = Inst().V.GetNodeForVolume(vols[0],
-						defaultCommandTimeout, defaultCommandRetry)
-					Expect(err).NotTo(HaveOccurred())
+					// The first 2 failovers are done by stopping PX on the NFS server node.
+					// The next 2 failovers are done by rebooting the node.
+					if i < 2 {
+						failoverMethod = volDriverStop
```

Comment:
Is there a reason you want to combine this in a single test and not have two separate tests? It would be more deterministic if each test does one kind of failure scenario.

Reply:
Main reason is to avoid duplicating the code. I experimented a bit using BeforeEach(), JustBeforeEach() etc But that did not seem to fit well here.